drydock/plugin.py

```diff
@@ -1,11 +1,140 @@
 from glob import glob
 import os
+import click
 import pkg_resources
 
-from tutor import hooks
+import typing as t
+
+import functools
+from .hooks import SYNC_WAVES_ORDER_ATTRS_TYPE, SYNC_WAVES_ORDER
+
+from tutor import hooks as tutor_hooks
+from tutor import env as tutor_env
+from tutor import serialize, types
+from tutor import config as tutor_config
 
 from .__about__ import __version__
 
+INIT_JOBS_SYNC_WAVE = 1
+
+# This function is taken from
+# https://github.com/overhangio/tutor/blob/v15.3.7/tutor/commands/k8s.py#L180
+def _load_jobs(tutor_conf: types.Config) -> t.Iterable[t.Any]:
+    jobs = tutor_env.render_file(tutor_conf, "k8s", "jobs.yml").strip()
+    for manifest in serialize.load_all(jobs):
+        if manifest["kind"] == "Job":
+            yield manifest
+
+
+# The definition of the init tasks is taken and adapted from
+# https://github.com/overhangio/tutor/blob/v15.3.7/tutor/commands/jobs.py#L64
+# and https://github.com/overhangio/tutor/blob/v15.3.7/tutor/commands/k8s.py#L80
+def get_init_tasks():
+    """Return the list of init tasks to run."""
+    init_tasks = list(tutor_hooks.Filters.CLI_DO_INIT_TASKS.iterate())
+    context = click.get_current_context().obj
+    tutor_conf = tutor_config.load(context.root)
+
+    # Standarize deprecated COMMANDS_INIT and COMMANDS_PRE_INIT Filter
+    pre_init_tasks = []
+    for service, init_path in tutor_hooks.Filters.COMMANDS_PRE_INIT.iterate():
+        pre_init_tasks.append((service, tutor_env.read_template_file(*init_path)))
+
+    init_tasks = pre_init_tasks + init_tasks
+
+    for service, init_path in list(tutor_hooks.Filters.COMMANDS_INIT.iterate()):
+        init_tasks.append((service, tutor_env.read_template_file(*init_path)))
+
+    for i, (service, command) in enumerate(init_tasks):
+        for template in _load_jobs(tutor_conf):
+            if template['metadata']['name'] != service + '-job':
+                continue
+
+            render_command = tutor_env.render_str(tutor_conf, command)
+
+            template['metadata']['name'] = f"drydock-{template['metadata']['name']}-{i}"
+            template['metadata']['labels'] = {
+                'drydock.io/component': 'job',
+                'drydock.io/target-service': template['metadata']['name'],
+                'drydock.io/runner-service': template['metadata']['name']
+            }
```

Comment:
This overrides the existing label, is that something we should be concerned? I think they only have the `app.kubernetes.io/component: job` label.

We could use `.update()` to merge the labels (we would probably need to override `app.kubernetes.io/component: job` because we have a patch to skip those).